### PR TITLE
clh: Support VFIO device unplug

### PR DESCRIPTION
**Pull request by likebreath**
Forward port PR kata-containers/runtime#2996 for issue kata-containers/runtime#2569.

This patch adds the support of VFIO device unplug when using
cloud-hypervisor.

Fixes: #860

Signed-off-by: Bo Chen chen.bo@intel.com

### CODE REVIEW

**Review by devimc — APPROVED** [review]
thanks @likebreath  - lgtm

VFIO test is passing but I need to disable DAX

see https://github.com/kata-containers/kata-containers/issues/862

File: src/runtime/virtcontainers/clh.go [hunk 1]
```diff
@@ -515,13 +507,29 @@ func (clh *cloudHypervisor) hotplugRemoveBlockDevice(drive *config.BlockDrive) e
...
 
+func (clh *cloudHypervisor) hotplugRemoveVfioDevice(device *config.VFIODev) error {
+	cl := clh.client()
+	ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+	defer cancel()
+
+	_, err := cl.VmRemoveDevicePut(ctx, chclient.VmRemoveDevice{Id: device.ID})
+
+	if err != nil {
+		err = fmt.Errorf("failed to hotplug remove vfio device %+v %s", device, openAPIClientError(err))
```
Comment (jodh-intel) [inline src/runtime/virtcontainers/clh.go:518]: Nit: I'd consider adding a colon just before the original error:

```suggestion
		err = fmt.Errorf("failed to hotplug remove vfio device %+v: %s", device, openAPIClientError(err))
```

**Review by jodh-intel — APPROVED** [review]
Thanks @likebreath.

lgtm

**Pull request merged**